Enh/testsuite temp folder

mschmidt87 opened the pull request:
This pull request does two things:

- It fixes #54 : The `get_previous_version` function is extended to take arbitrary paths. The test in `test_wrapper.py` is modified to use the tmpdir functionality of pytest. The test now loads the previous version and stores it in the temporary folder of the system.

- The conversion test should always test the current version against the previous version. This previous version is now 1.0.1., so I updated this in this pull-request as well.

jakobj commented on h5py_wrapper/lib.py:
```diff
@@ -10,25 +10,37 @@
+    base_url = "https://github.com/INM-6/h5py_wrapper/archive/"
     r = requests.get(''.join((base_url, version, ".tar.gz")))
+    # Convert path to str
+    path = str(path)
```
why do we need this?

mschmidt87 replied:
Because before, `path` is a `LocalPath` object created by the mechanism of `py.test` for temporary directories. This object cannot be handled by `os.path.join` which only accepts `str` objects. There is a way to join `LocalPath` objects but I think handling the path as string is more general and easier here.

jakobj replied:
OK. Could you clarify that in the comment? `convert LocalPath object to str` or something like that?

jakobj commented on h5py_wrapper/lib.py:
```diff
@@ -10,25 +10,37 @@
-        os.remove(fn)
+            f.extractall(path=path)
+        # This is necessary to account for the fact that version 0.0.1 was called v0.0.1
+        # but all subsequent version are called X.X.X without the leading v.
```
ouch, bad call from us on changing the convention. would it be possible to fix this retrospectively by modifying the release? but I guess this causes more problems than it solves.

mschmidt87 replied:
I don't think that it is possible. After all, it's the purpose of a release that it can't be changed, isn't it?

jakobj replied:
true. let's not mess with that. we know better next time.

jakobj commented on tests/test_wrapper.py:
```diff
@@ -319,28 +319,29 @@ def test_file_close_on_exception():
+def test_conversion_script(tmpdir):
     try:
-        import h5py_wrapper_001.wrapper as h5w_001
+        import h5py_wrapper_101.wrapper as h5w_101
```
why did the version change from 001 to 101 in this PR?

mschmidt87 replied:
Originally, we wrote this test after releasing 0.0.1. After releasing 1.0.1., we did not change this here, although I think that we should have, because in my understanding, this test should test against the previous release. Or against all previous releases?

jakobj replied:
I think we should provide conversion from alll old versions to the current version, so this test should check for all previous releases. Is this possible with small changes? Otherwise we should handle this in a new PR.

mschmidt87 commented:
I now implemented a loop over all previous version in `test_conversion_script`. I also implemented the temporary directory in `convert_h5`. I had to adapt my previous solution a little bit to account for the fact that we changed the directory structure of the module from 0.0.1 to 1.0.1 (`wrapper.py` was moved one level down to `h5py_wrapper/wrapper.py`.

Note that with this PR, `test_conversion_script` currently fails because of a bug I discovered. This bug is fixed in .

jakobj commented:
Looks good :+1:  and merge
